generic_scripts/nlk.py

```diff
@@ -0,0 +1,290 @@
+                wire_positions[i+1] = \
+                    _np.array([s1[j]*10., s2[j]*5.85])[None, :]*1e-3
+
+        # Creating wires
+        self._wires = []
+        for i in range(8):
+            if i % 2:
+                curr = -current
+            else:
+                curr = current
+            pos = wire_positions[i]
+            wire = Wire(pos, curr)
+            self._wires.append(wire)
+
+    @property
+    def wires(self):
+        """List with the Wires objects that compose the NLK.
+
+        Returns
+        -------
+        list of Wires objects.
+        """
+        return self._wires
+
+    @wires.setter
+    def wires(self, wires_objects):
+        """."""
+        self._wires = wires_objects
+
+    @property
+    def positions(self):
+        """Position coordinates of the wires.
+
+        Returns
+        -------
+        numpy.array
+            Returns a numpy.array of shape (2, 8) with the wires coordinates.
+        """
+        wires_positions = _np.zeros([2, 8])
+        for i in range(wires_positions.shape[1]):
+            wires_positions[:, [i]] = self._wires[i].position[:, None]
+        return wires_positions
+
+    @positions.setter  # Thinking in a good setter for this...
+    def positions(self, wires_positions):
+        for i in range(wires_positions.shape[1]):
+            self._wires[i].position = wires_positions[:, i]
+
+    @property
+    def currents(self):
+        """Currents of the NLK wires in [A].
+
+        Returns
+        -------
+        numpy.array
+            Single dimension array with the currents of each wire in NLK.
+        """
+        wires_currents = _np.zeros(len(self._wires))
+        for i in range(len(self._wires)):
+            wires_currents[i] = self._wires[i].current
+        return wires_currents
+
+    @currents.setter
+    def currents(self, wires_currents):
+        for i, current in enumerate(wires_currents):
+            self._wires[i].current = current
+
+    def calc_magnetic_field(self, r):
+        """Computes the magnetic field generated by NLK in a set of points.
+
+        Parameters
+        ----------
+        r : numpy.array
+            Array of shape (2 x N) with the coordinates [[x1, x2, ... , xN],
+            [y1, y2, ... , yN]] where the field will be returned.
+
+        Returns
+        -------
+        numpy.array
+            Array with the same shape of r, containing the magnetic field
+            components in [T] at each point in r.
+
+        Raises
+        ------
+        ValueError
+            If the input r hasn't the shape of type (2 X N).
+        """
+        if r.shape[0] != 2:
+            raise ValueError(
+                "The input position vector r must have shape of type (2, N)")
+        mag_field = _np.zeros(r.shape)
+        for wire in self.wires:
+            mag_field += wire.calc_magnetic_field(r)
+        return mag_field
+
+    def get_magnetic_field_on_axis(self):
```

Comment:
why not have a more generic method `get_vertical_magnetic_field` with `y` input defining the plane, with default value to be 0? And also have `get_horizontal_magnetic_field` with `x` input also defining a plane.

Reply:
It's a good idea, I will implement it.